assets/src/components/v2/departures/normal_section.tsx

```diff
@@ -19,9 +21,15 @@ type NormalSection = {
   layout: Layout;
   header: React.ComponentProps<typeof Header>;
   rows: Row[];
+  trimmedRows?: Row[];
```

Comment:
`trimmedRows` makes sense as a name for the output of the "trimming" process, but from the perspective of this component, which isn't really involved in that, it feels odd to me. Maybe `laterRows`? (to align with `include_later` — if we "include later", then we include the "later rows")

Also, if there's a way we could make this non-optional, it seems like it would simplify some of the logic here and in the trimming process, especially since there's not a meaningful difference between the key being absent and being set to `[]`.

Reply:
> Also, if there's a way we could make this non-optional, it seems like it would simplify some of the logic here and in the trimming process, especially since there's not a meaningful difference between the key being absent and being set to `[]`

The challenge here is that this type is used as both a value for React component props and as a representation of API response data. The `trimSections` function transforms this from one type to another. If we separated API data types from component props types we could easily have `trimSections` return the latter or just do this operation from `APINormalSection` -> `ClientNormalSection` before running `trimSections` so `trimSections` is only working on `ClientNormalSection`.

I guess what I'm trying to say is that removing "optional" from this key passes the type checker but is actually not an accurate representation of the data flowing through the system.

Would you feel okay about leaving this for now and revisiting how we handle data from t he API and its types more holistically?

Reply:
> The `trimSections` function transforms this from one type to another. If we separated API data types from component props types

We do already in some places, for example `DepartureTimes`, and I think it would be a reasonable approach here. If you think that separation will be too hairy to perform as part of this task, it's probably fine, I just think it could be confusing that (as you say) we're conceptually going from one distinct type to another but using the same type in both cases.